x-pack/legacy/plugins/actions/server/init.ts

```diff
@@ -75,6 +81,9 @@ export function init(server: Legacy.Server) {
   const fireFn = createFireFunction({
```

Comment:
`fireFn`, which we expose via `server.expose`, allows consumers to specify their own space id. Do we know of any scenarios where consumers will need this capability? If not, we could potentially create a "scoped" fire function, where the space id is derived from the current request, or similar.

If we do need to allow consumers to specify their own space id, then you'll likely want to update [this](https://github.com/mikecote/kibana/blob/896b6a4b111b3ad582403c84808bb6d7904d0a07/x-pack/legacy/plugins/actions/server/create_fire_function.ts#L29) so that it can handle spaces being disabled, and handle cases where a space id is required, but missing. Currently, `spaceIdToNamespace` will throw an error if a space id is not provided.

Also, the underlying `fireFn` uses `callWithInternalUser` in an unguarded fashion, so there isn't any authorization being applied there. This would be less of a concern if it wasn't exposed to consumers, but since this is a consumer-facing function, we should have protections in place there. (perhaps out of scope for this PR?)

Reply:
From what I can tell, consumers will need to specify their own space id. The example I can think of is task manager wants to fire action `a` in space `b` and uses this function to indicate that. We should be ok when spaces is disabled due to `spaceIdToNamespace` being passed as https://github.com/elastic/kibana/pull/42081/files#diff-c2af122155b3536ee79e8b5384fdaa6eR94.

In regards to `callWithInternalUser` being unguarded, we'll have this fixed when API keys are implemented. We will do the `callWithInternalUser` first, setup a connection with the keys and re-load the same object via the saved objects client.